docs/proposals/002-api-proposal/proposal.md

```diff
@@ -183,147 +181,131 @@ type TargetModel struct {
         // and will result in sub-optimal utilization.
         ModelServerSelector map[string]string `json:"modelServerSelector,omitempty"`
 }
 ```
 
 ### Yaml Examples
 
-#### LLMServerPool(s)
-Here we create 2 LSPs that subscribe to services to collect the appropriate pods
+#### InferencePool(s)
+Here we create a pool that selects the appropriate pods
 ```yaml
 apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMServerPool
+kind: InferencePool
 metadata:
-  name: llama-2-pool
-  services: 
-  - llama-2-vllm
----
-apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMServerPool
-metadata:
-  name: gemini-pool
-  services: 
-  - gemini-jetstream-tpu-v5e
-  - gemini-vllm-a100
+  name: base-model-pool
+  modelServerSelector:
+  - app: llm-server
 ```
 
-#### LLMService
+#### InferenceModel
 
-Here we consume both pools with a single LLMService, while also specifying 2 LLMServices. Where `sql-code-assist` is both the name of the ModelLLMService, and the name of the LoRA adapter on the model server. And `npc-bot` has a layer of indirection for those names, as well as a specified objective. Both `sql-code-assist` and `npc-bot` have available LoRA adapters on both LLMServerPools and routing to each LLMServerPool happens earlier(at the K8s Gateway). So traffic splitting between separate pools happens at the K8s Gateway.
+Here we consume the pool with two InferenceModels. Where `sql-code-assist` is both the name of the model and the name of the LoRA adapter on the model server. And `npc-bot` has a layer of indirection for those names, as well as a specified criticality. Both `sql-code-assist` and `npc-bot` have available LoRA adapters on the InferencePool and routing to each InferencePool happens earlier (at the K8s Gateway).
 ```yaml
 apiVersion: inference.x-k8s.io/v1alpha1
-kind: LLMService
+kind: InferenceModel
+metadata:
+  name: sql-code-assist
+spec:
+  modelName: sql-code-assist
+  poolRef: base-model-pool
+---
+apiVersion: inference.x-k8s.io/v1alpha1
+kind: InferenceModel
 metadata:
-  name: my-llm-service
+  name: npc-bot
 spec:
-  LLMServices:
-  - modelName: sql-code-assist
-  - modelName: npc-bot
-    objective: 
-      desiredAveragePerOutputTokenLatencyAtP95OverMultipleRequests: 50ms
-    targetModels:
-      targetModelName: npc-bot-v1
-        weight: 50
-      targetModelName: npc-bot-v2
-        weight: 50 	
-  poolRef: 
-   - name: llama-2-pool
-   - name: gemini-pool
+  modelName: npc-bot
+  criticality: Critical
+  targetModels:
+    targetModelName: npc-bot-v1
+      weight: 50
+    targetModelName: npc-bot-v2
+      weight: 50 	
+  poolRef: base-model-pool
 ```
 
 ### Diagrams
 
 Much of this is better explained visually:
 
-Below is a detailed view of the LLMServerPool
+Below is a detailed view of the InferencePool
 
-![LLMServerPool](./images/lsp.svg)
+![InferencePool](./images/lsp.svg)
 
 This diagram lightly follows the example request for a model `name-generator`. 
```

Comment:
It will be better to use the same pool and model names in the diagram as the example yaml above.